src/memray/reporters/flamegraph.py

```diff
@@ -83,22 +216,33 @@ def _from_any_snapshot(
         memory_records: Iterable[MemorySnapshot],
         native_traces: bool,
         temporal: bool,
+        inverted: Optional[bool] = None,
     ) -> "FlameGraphReporter":
-        root: Dict[str, Any] = {
+        root: FrameNodeDict = {
             "name": "<root>",
-            "location": [html.escape("<tracker>"), "<b>memray</b>", 0],
+            "location": (html.escape("<tracker>"), "<b>memray</b>", 0),
             "value": 0,
             "children": [],
             "n_allocations": 0,
             "thread_id": "0x0",
             "interesting": True,
             "import_system": False,
         }
+        inverted = False if inverted is None else inverted
+
+        frames: List[FrameNodeDict] = [root]
+        inverted_no_imports_frames: List[FrameNodeDict] = []
+
+        if inverted:
+            inverted_no_imports_root = deepcopy(root)
+            inverted_no_imports_frames = [inverted_no_imports_root]
 
-        frames = [root]
         interval_list: List[Tuple[int, Optional[int], int, int, int]] = []
 
         node_index_by_key: Dict[Tuple[int, StackFrame, str], int] = {}
+        inverted_no_imports_node_index_by_key: Dict[
+            Tuple[int, StackFrame, str], int
+        ] = {}
```

Comment:
I'm gonna switch this to use a type alias, to make things slightly easier to read:
```py
        NodeKey = Tuple[int, StackFrame, str]
        node_index_by_key: Dict[NodeKey, int] = {}
        inverted_no_imports_node_index_by_key: Dict[NodeKey, int] = {}
```
In particular, it's tough to see that the two variables have the same type given the way that Black wrapped the lines, and this change makes that more recognizable.